application/src/main/java/org/togetherjava/tjbot/commands/reminder/ReminderCommand.java

```diff
@@ -0,0 +1,297 @@
+    }
+
+    private MessageCreateData createListMessage(Result<PendingRemindersRecord> pendingReminders,
+            int pageToShow) {
+        int totalPages = (int) Math.ceil(pendingReminders.size() / (double) MAX_PAGE_LENGTH);
+
+        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
+            .setColor(RemindRoutine.AMBIENT_COLOR);
+        MessageCreateBuilder listMessage = new MessageCreateBuilder();
+
+        if (pendingReminders.isEmpty()) {
+            remindersEmbed.setDescription("No pending reminders");
+        } else {
+            if (totalPages > 1) {
+                remindersEmbed.setFooter("Page: " + pageToShow + "/" + totalPages);
+
+                getRemindersPage(pendingReminders, pageToShow)
+                    .forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+
+                listMessage.addActionRow(createPageTurnButtons(pageToShow, totalPages));
+            } else {
+                pendingReminders.forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+            }
+        }
+
+        return listMessage.addEmbeds(remindersEmbed.build()).build();
+    }
+
+    private List<Button> createPageTurnButtons(int pageNumber, int totalPages) {
+        int previousButtonTurnPageBy = -1;
+        Button previousButton =
+                createPageTurnButton(PREVIOUS_BUTTON_LABEL, pageNumber, previousButtonTurnPageBy);
+        if (pageNumber <= 1) {
+            previousButton = previousButton.asDisabled();
+        }
+
+        int nextButtonTurnPageBy = 1;
+        Button nextButton =
+                createPageTurnButton(NEXT_BUTTON_LABEL, pageNumber, nextButtonTurnPageBy);
+        if (pageNumber >= totalPages) {
+            nextButton = nextButton.asDisabled();
+        }
+
+        return List.of(previousButton, nextButton);
+    }
+
+    private Button createPageTurnButton(String label, long pageNumber, int turnPageBy) {
+        return Button.primary(
+                generateComponentId(String.valueOf(pageNumber), String.valueOf(turnPageBy)), label);
+    }
+
+    private static List<PendingRemindersRecord> getRemindersPage(
+            Result<PendingRemindersRecord> remindersRecords, int pageNumber) {
+        List<PendingRemindersRecord> reminders = new ArrayList<>(MAX_PAGE_LENGTH);
+
+        int start = (pageNumber - 1) * MAX_PAGE_LENGTH;
+        int end = Math.min(start + MAX_PAGE_LENGTH, remindersRecords.size());
+
+        for (int i = start; i < end; i++) {
+            reminders.add(remindersRecords.get(i));
+        }
+
+        return reminders;
+    }
+
+    private static void addReminderAsField(PendingRemindersRecord reminder, EmbedBuilder embed) {
+        BiFunction<Long, Instant, String> getDescription = (channelId, remindAt) -> """
+                Channel: %s
+                Remind at: %s""".formatted(MessageUtils.mentionChannelById(channelId),
+                TimeFormat.DEFAULT.format(remindAt));
```

Comment:
Getting back to the topic of timezones, you dont need to format time yourself. Discord added timestamp formatting and the time gets adjusted for the users timezone. How cool! Here is some info on it: https://gist.github.com/LeviSnoot/d9147767abeef2f770e9ddcd91eb85aa

Reply:
that is exactly what this method does, first i did that manually but zabu said theres a method for that in jda

Reply:
Oh ok. I thought TimeFormat was a standard java class